In [ ]:
#IMPORT LIBRARY
import numpy as np
import pandas as pd
import scipy.stats as stats 


In [ ]:
# Mengimport Data 
from google.colab import files
uploaded = files.upload()

In [ ]:
# Read Data yang sudah di upload
df = pd.read_csv('/content/marketing_campaign.csv', sep='\t')
df


ID  Year_Birth   Education Marital_Status   Income  Kidhome  \
0      5524        1957  Graduation         Single  58138.0        0   
1      2174        1954  Graduation         Single  46344.0        1   
2      4141        1965  Graduation       Together  71613.0        0   
3      6182        1984  Graduation       Together  26646.0        1   
4      5324        1981         PhD        Married  58293.0        1   
...     ...         ...         ...            ...      ...      ...   
2235  10870        1967  Graduation        Married  61223.0        0   
2236   4001        1946         PhD       Together  64014.0        2   
2237   7270        1981  Graduation       Divorced  56981.0        0   
2238   8235        1956      Master       Together  69245.0        0   
2239   9405        1954         PhD        Married  52869.0        1   

      Teenhome Dt_Customer  Recency  MntWines  ...  NumWebVisitsMonth  \
0            0  04-09-2012       58       635  ...                  7   
1            1  08-03-2014       38        11  ...                  5   
2            0  21-08-2013       26       426  ...                  4   
3            0  10-02-2014       26        11  ...                  6   
4            0  19-01-2014       94       173  ...                  5   
...        ...         ...      ...       ...  ...                ...   
2235         1  13-06-2013       46       709  ...                  5   
2236         1  10-06-2014       56       406  ...                  7   
2237         0  25-01-2014       91       908  ...                  6   
2238         1  24-01-2014        8       428  ...                  3   
2239         1  15-10-2012       40        84  ...                  7   

      AcceptedCmp3  AcceptedCmp4  AcceptedCmp5  AcceptedCmp1  AcceptedCmp2  \
0                0             0             0             0             0   
1                0             0             0             0             0   
2                0             0             0             0             0   
3                0             0             0             0             0   
4                0             0             0             0             0   
...            ...           ...           ...           ...           ...   
2235             0             0             0             0             0   
2236             0             0             0             1             0   
2237             0             1             0             0             0   
2238             0             0             0             0             0   
2239             0             0             0             0             0   

      Complain  Z_CostContact  Z_Revenue  Response  
0            0              3         11         1  
1            0              3         11         0  
2            0              3         11         0  
3            0              3         11         0  
4            0              3         11         0  
...        ...            ...        ...       ...  
2235         0              3         11         0  
2236         0              3         11         0  
2237         0              3         11         0  
2238         0              3         11         0  
2239         0              3         11         1  

[2240 rows x 29 columns]

In [ ]:
#Menampilkan 5 Data Teratas
df.head()

ID  Year_Birth   Education Marital_Status   Income  Kidhome  Teenhome  \
0  5524        1957  Graduation         Single  58138.0        0         0   
1  2174        1954  Graduation         Single  46344.0        1         1   
2  4141        1965  Graduation       Together  71613.0        0         0   
3  6182        1984  Graduation       Together  26646.0        1         0   
4  5324        1981         PhD        Married  58293.0        1         0   

  Dt_Customer  Recency  MntWines  ...  NumWebVisitsMonth  AcceptedCmp3  \
0  04-09-2012       58       635  ...                  7             0   
1  08-03-2014       38        11  ...                  5             0   
2  21-08-2013       26       426  ...                  4             0   
3  10-02-2014       26        11  ...                  6             0   
4  19-01-2014       94       173  ...                  5             0   

   AcceptedCmp4  AcceptedCmp5  AcceptedCmp1  AcceptedCmp2  Complain  \
0             0             0             0             0         0   
1             0             0             0             0         0   
2             0             0             0             0         0   
3             0             0             0             0         0   
4             0             0             0             0         0   

   Z_CostContact  Z_Revenue  Response  
0              3         11         1  
1              3         11         0  
2              3         11         0  
3              3         11         0  
4              3         11         0  

[5 rows x 29 columns]

In [ ]:
#Menampilkan Data Secara Acak
df.sample(5)

ID  Year_Birth   Education Marital_Status   Income  Kidhome  \
1035   9503        1980  Graduation         Single  28071.0        0   
1715  10340        1984  Graduation       Together  35196.0        1   
834   10402        1967  Graduation        Married  35441.0        1   
1249  10833        1975  Graduation        Married  65106.0        0   
1684  10323        1952  Graduation        Married  49413.0        0   

      Teenhome Dt_Customer  Recency  MntWines  ...  NumWebVisitsMonth  \
1035         0  29-08-2012       65        39  ...                  8   
1715         0  13-11-2012       68        75  ...                  8   
834          1  14-02-2013       94        25  ...                  8   
1249         1  24-05-2014       55       790  ...                  6   
1684         1  25-10-2013       88       205  ...                  5   

      AcceptedCmp3  AcceptedCmp4  AcceptedCmp5  AcceptedCmp1  AcceptedCmp2  \
1035             0             0             0             0             0   
1715             0             0             0             0             0   
834              0             0             0             0             0   
1249             0             1             0             0             0   
1684             0             0             0             0             0   

      Complain  Z_CostContact  Z_Revenue  Response  
1035         0              3         11         0  
1715         0              3         11         0  
834          0              3         11         0  
1249         0              3         11         0  
1684         0              3         11         0  

[5 rows x 29 columns]

In [ ]:
#Mengecek fitur pada kolom yang memiliki data kosong
df.isnull().sum()

ID                      0
Year_Birth              0
Education               0
Marital_Status          0
Income                 24
Kidhome                 0
Teenhome                0
Dt_Customer             0
Recency                 0
MntWines                0
MntFruits               0
MntMeatProducts         0
MntFishProducts         0
MntSweetProducts        0
MntGoldProds            0
NumDealsPurchases       0
NumWebPurchases         0
NumCatalogPurchases     0
NumStorePurchases       0
NumWebVisitsMonth       0
AcceptedCmp3            0
AcceptedCmp4            0
AcceptedCmp5            0
AcceptedCmp1            0
AcceptedCmp2            0
Complain                0
Z_CostContact           0
Z_Revenue               0
Response                0
dtype: int64

In [ ]:
#Mengisi Data Kosong
df['Income'] = df['Income'].fillna(df['Income'].median())
print(df['Income'])

0       58138.0
1       46344.0
2       71613.0
3       26646.0
4       58293.0
         ...   
2235    61223.0
2236    64014.0
2237    56981.0
2238    69245.0
2239    52869.0
Name: Income, Length: 2240, dtype: float64


In [ ]:
#Mengecek Kembali Data Kosong
df.isnull().sum()

ID                     0
Year_Birth             0
Education              0
Marital_Status         0
Income                 0
Kidhome                0
Teenhome               0
Dt_Customer            0
Recency                0
MntWines               0
MntFruits              0
MntMeatProducts        0
MntFishProducts        0
MntSweetProducts       0
MntGoldProds           0
NumDealsPurchases      0
NumWebPurchases        0
NumCatalogPurchases    0
NumStorePurchases      0
NumWebVisitsMonth      0
AcceptedCmp3           0
AcceptedCmp4           0
AcceptedCmp5           0
AcceptedCmp1           0
AcceptedCmp2           0
Complain               0
Z_CostContact          0
Z_Revenue              0
Response               0
dtype: int64

In [ ]:
df['Income'].describe()


count      2240.000000
mean      52237.975446
std       25037.955891
min        1730.000000
25%       35538.750000
50%       51381.500000
75%       68289.750000
max      666666.000000
Name: Income, dtype: float64

**# Descriptive Statistics**

# Measure of Central Tendency

In [ ]:
#Mencari Rata-rata
Income = df['Income']
print('Mean:', np.mean(Income))

Mean: 52237.97544642857


In [ ]:
#Mencari Nilai Tengah
Income = df['Income']
print('Median',np.median(Income))

Median 51381.5


In [ ]:
#Mencari Modus
sat_mode = df['Income'].mode()[0] 

In [ ]:
sat_mode

51381.5



#Measure of Spread

In [ ]:
#Mencari Variance
sat_var = df['Income'].var()
sat_sd = df['Income'].std()
print('variance:',sat_var)
print('standard deviation',sat_sd)


variance: 626899235.1807307
standard deviation 25037.955890621957


#Interpretasi
Nilai dari measure of tendency dari income adalah sebagai berikut:


 Mean: 52237.97544642857

 Median: 51381.5

 Modus: 51381.5




 Nilai dari measure of spread dari income adalah sebagai berikut:


 variance: 626899235.1807307

 standard deviation: 25037.955890621957

**# Hypothesis Testing**

In [ ]:
#Menggabungkan Fitur Kolom Education dan Income untuk menganalisis Uji Hipotesis (cara 1)
Edu_Income = df.groupby(['Education'])['Income']
Edu_Income.describe()

count          mean           std     min       25%      50%  \
Education                                                                   
2n Cycle     203.0  47688.583744  21958.898270  7500.0  28095.00  46891.0   
Basic         54.0  20306.259259   6235.066773  7500.0  15405.25  20744.0   
Graduation  1127.0  52707.305679  28039.531293  1730.0  34925.50  51651.0   
Master       370.0  52896.777027  20021.539473  6560.0  37853.75  51044.5   
PhD          486.0  56096.303498  20512.097287  4023.0  40654.25  55005.0   

                75%       max  
Education                      
2n Cycle    65100.5   96547.0  
Basic       24882.0   34445.0  
Graduation  69737.0  666666.0  
Master      66596.0  157733.0  
PhD         69084.0  162397.0

In [ ]:
df.groupby(['Education'])['Income'].describe() # (cara 2)

count          mean           std     min       25%      50%  \
Education                                                                   
2n Cycle     203.0  47688.583744  21958.898270  7500.0  28095.00  46891.0   
Basic         54.0  20306.259259   6235.066773  7500.0  15405.25  20744.0   
Graduation  1127.0  52707.305679  28039.531293  1730.0  34925.50  51651.0   
Master       370.0  52896.777027  20021.539473  6560.0  37853.75  51044.5   
PhD          486.0  56096.303498  20512.097287  4023.0  40654.25  55005.0   

                75%       max  
Education                      
2n Cycle    65100.5   96547.0  
Basic       24882.0   34445.0  
Graduation  69737.0  666666.0  
Master      66596.0  157733.0  
PhD         69084.0  162397.0

In [ ]:
#Metode Pengujian Hypothesis menggunakan Anove

dfCycle = df[df['Education'] == '2n Cycle']
dfBasic = df[df['Education'] == 'Basic']
dfGraduation = df[df['Education'] == 'Graduation']
dfMaster = df[df['Education'] == 'Master']
dfPhD = df[df['Education'] == 'PhD']

In [ ]:
anove_test = stats.f_oneway(dfCycle['Income'],dfBasic['Income'],dfGraduation['Income'],dfMaster['Income'],dfPhD['Income'])

In [ ]:
result = anove_test.pvalue

In [ ]:
if result > 0.5 :
  print('Education tidak berpengaruh terhadap Income')
else:
  print('Education berpengaruh terhadap Income')

NameError: ignored

Hasil Hypothesis Testing menunjukkan bahwa Education berpengaruh terhadap Income